spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/convert/TwoStepsConversions.java

```diff
@@ -148,11 +174,19 @@ public Value convertOnWrite(Object proppertyVal) {
 		if (val instanceof Iterable) {
 			List<Value<?>> values = new ArrayList<>();
 			for (Object propEltValue : (Iterable) val) {
-				values.add(this.convertOnWriteSingle(propEltValue));
+				values.add(writeConverter.apply(propEltValue));
 			}
 			return ListValue.of(values);
 		}
-		return convertOnWriteSingle(val);
+		return writeConverter.apply(val);
+	}
+
+	private EntityValue convertSingleEmbeddedEntity(Object val) {
+		IncompleteKey key = this.objectToKeyFactory.getIncompleteKey("embedded");
```

Comment:
"embedded" is the kind name here for this fake key. I think you should make this a static final member somewhere. Also, the name "embedded" seems a little too common, and might be used for real as a kind name, don't you think? 

What if it was like "spring-data-cloud-datastore-embedded-dummy-kind-name" so that it will never be used. 

Finally , it can definitely work that the embedded class type is actually a real entity with a real type, but maybe we don't want to support this because it creates the opportunity for misuse and confusion.